app/logic/adminNewEvent.py

```diff
@@ -24,32 +52,58 @@ def createNewEvent(newEventData):
+                                                  timeStart = newEventData['eventStartTime'],
+                                                  timeEnd = newEventData['eventEndTime'],
+                                                  location = newEventData['eventLocation'],
+                                                  isRecurring = newEventData['eventIsRecurring'],
+                                                  isRsvpRequired = newEventData['eventRSVP'],
+                                                  isPrerequisiteForProgram = newEventData['eventRequiredForProgram'],
+                                                  isTraining = newEventData['eventIsTraining'],
+                                                  isService = newEventData['eventServiceHours'],
+                                                  startDate =  newEventData[f"week{entry}"],
+                                                  endDate =  newEventData[f"week{entry}"],
+                                                  program = newEventData['programId'])
+
+                        programEvent = ProgramEvent.create(program=program, event=newEvent)
+
+                        facilitatorEntry = Facilitator.create(user = newEventData['eventFacilitator'],
+                                                                  event = newEvent)
+                else:
+                    break
+
+        else:
+            with mainDB.atomic():
+                newEvent = Event.create(eventName = newEventData['eventName'],
+                                          term = newEventData['eventTerm'],
+                                          description= newEventData['eventDescription'],
+                                          timeStart = newEventData['eventStartTime'],
+                                          timeEnd = newEventData['eventEndTime'],
+                                          location = newEventData['eventLocation'],
+                                          isRecurring = newEventData['eventIsRecurring'],
+                                          isRsvpRequired = newEventData['eventRSVP'],
+                                          isPrerequisiteForProgram = newEventData['eventRequiredForProgram'],
+                                          isTraining = newEventData['eventIsTraining'],
+                                          isService = newEventData['eventServiceHours'],
+                                          startDate =  newEventData['eventStartDate'],
+                                          endDate =  newEventData['eventEndDate'],
+                                          program = newEventData['programId'])
+
+                programEvent = ProgramEvent.create(program=program, event=newEvent)
+
+                facilitatorEntry = Facilitator.create(user = newEventData['eventFacilitator'],
+                                          event = newEvent)
+

```

Comment:
Since the `else` case is the same as the main case, I would combine them. You just need to only loop once and don't add the "Week X" string, and they are the same. Also, program shouldn't be in the event.create() list - it probably was added back in during conflict resolution.